src/compiler/generate-common-tests.xsl

```diff
@@ -784,12 +786,17 @@
       <xsl:variable name="qname" as="xs:QName"
          select="x:resolve-EQName-ignoring-default-ns(@name, .)" />
 
-      <xsl:if test="namespace-uri-from-QName($qname) eq $xspec-namespace">
-         <xsl:variable name="msg" as="xs:string">
-            <xsl:text expand-text="yes">User-defined XSpec variable, {@name}, must not use the XSpec namespace.</xsl:text>
-         </xsl:variable>
-         <xsl:sequence select="xs:QName('x:XSPEC008') => error($msg)" />
-      </xsl:if>
+      <xsl:choose>
+         <xsl:when test="$is-external and ($qname eq xs:QName('x:saxon-config'))">
+            <!-- Allow it -->
```

Comment:
Your suggestion for a future element name sounds good. I added a comment in 8bedfddcc03d4cf2778bdd933ace5ee0cd3cb85d so that we don't forget the point.

Regarding the Saxon configuration, [`SAXON_CUSTOM_OPTIONS`](https://github.com/xspec/xspec/search?q=SAXON_CUSTOM_OPTIONS&type=Wikis) and [`saxon.custom.options`](https://github.com/xspec/xspec/search?q=saxon.custom.options&type=Wikis) provides a similar feature, although it is always global and cannot be scenario-specific.

So the use cases of `run-as="external"` I'm aware of are

- XSLT packages (#762)
- Source node (#873)
- Scenario-specific Saxon configuration

Reply:
I added **Possible use cases** section to this pull request's [description](https://github.com/xspec/xspec/pull/892#issue-396249022).

Reply:
Thanks for the note about `SAXON_CUSTOM_OPTIONS` and `saxon.custom.options`. Good distinction about those being global and this new feature offering scenario-specific control.